File: backend/src/test/java/com/pokedex/backend/pokemon/PokemonFixtures.java
package com.pokedex.backend.pokemon;

import java.util.List;
import java.util.Map;

final class PokemonFixtures {

	private PokemonFixtures() {
	}

	static Pokemon persist(
			PokemonRepository repository,
			int pokeapiId,
			int nationalDexId,
			String name,
			List<String> types,
			String generation,
			boolean defaultVariant) {
		Pokemon pokemon = new Pokemon();
		pokemon.setPokeapiId(pokeapiId);
		pokemon.setNationalDexId(nationalDexId);
		pokemon.setName(name);
		pokemon.setHeightDecimeters(7);
		pokemon.setWeightHectograms(69);
		pokemon.setBaseExperience(64);
		pokemon.setTypes(types);
		pokemon.setAbilities(List.of(new PokemonAbility("overgrow", false)));
		pokemon.setStats(Map.of("hp", 45, "attack", 49));
		pokemon.setSprites(Map.of("official_artwork", "https://example.com/" + name + ".png"));
		pokemon.setGeneration(generation);
		pokemon.setDefaultVariant(defaultVariant);
		pokemon.setVariantOfNationalDexId(defaultVariant ? null : nationalDexId);
		return repository.saveAndFlush(pokemon);
	}

	static void seedCatalog(PokemonRepository repository) {
		persist(repository, 1, 1, "bulbasaur", List.of("grass", "poison"), "generation-i", true);
		persist(repository, 6, 6, "charizard", List.of("fire", "flying"), "generation-i", true);
		persist(repository, 25, 25, "pikachu", List.of("electric"), "generation-i", true);
		persist(repository, 152, 152, "chikorita", List.of("grass"), "generation-ii", true);
		persist(repository, 10035, 6, "charizard-mega-x", List.of("fire", "dragon"), "generation-vi", false);
	}
}

File: pokedex-app/src/test/pokemonFixtures.ts
import type { Pokemon } from '../services/pokeAPI.type';

export function makePokemon(overrides: Partial<Pokemon> = {}): Pokemon {
	return {
		id: 1,
		pokeapiId: 1,
		nationalDexId: 1,
		name: 'bulbasaur',
		heightDecimeters: 7,
		weightHectograms: 69,
		baseExperience: 64,
		types: ['grass', 'poison'],
		abilities: [{ name: 'overgrow', hidden: false }],
		stats: {
			hp: 45,
			attack: 49,
			defense: 49,
			'special-attack': 65,
			'special-defense': 65,
			speed: 45,
		},
		sprites: {
			official_artwork: 'https://example.com/bulbasaur.png',
		},
		generation: 'generation-i',
		defaultVariant: true,
		variantOfNationalDexId: null,
		createdAt: '2026-01-01T00:00:00Z',
		updatedAt: '2026-01-01T00:00:00Z',
		...overrides,
	};
}

export const samplePokemon = [
	makePokemon(),
	makePokemon({
		id: 6,
		pokeapiId: 6,
		nationalDexId: 6,
		name: 'charizard',
		types: ['fire', 'flying'],
		sprites: { official_artwork: 'https://example.com/charizard.png' },
	}),
	makePokemon({
		id: 25,
		pokeapiId: 25,
		nationalDexId: 25,
		name: 'pikachu',
		types: ['electric'],
		sprites: { official_artwork: 'https://example.com/pikachu.png' },
	}),
];
